### PR TITLE
Added other permission types to create open extensions in each resource.

### PR DESCRIPTION
**Pull request by angelgolfer-ms**

### CODE REVIEW

File: api-reference/v1.0/api/opentypeextension_delete.md [hunk 1]
```diff
@@ -4,15 +4,19 @@ Delete an open extension ([openTypeExtension](../resources/openTypeExtension.md)
 
 ## Permissions
 
-One of the following permissions is required to call this API, depending on the resource you're deleting the extension from. To learn more, including how to choose permissions, see [Permissions](../../../concepts/permissions_reference.md).
+Depending on the resource you're deleting the extension from and the permission type (delegated or application) requested, the specified permission in the following table is the mimimum required to call this API. To learn more, including how to choose permissions, see [Permissions](../../../concepts/permissions_reference.md).
```
Comment (dkershaw10) [inline api-reference/v1.0/api/opentypeextension_delete.md]: minimum required -> least privileged permission required OR  least privileged required
NOTE you might be able to get away without

File: api-reference/v1.0/api/opentypeextension_delete.md [hunk 2]
```diff
@@ -4,15 +4,19 @@ Delete an open extension ([openTypeExtension](../resources/openTypeExtension.md)
...
-| [user](../resources/user.md) | Directory.AccessAsUser.All | | |
+| [device](../resources/device.md) | Directory.AccessAsUser.All | Not supported | Device.ReadWrite.All |
+| [event](../resources/event.md) | Calendars.ReadWrite | Calendars.ReadWrite | Calendars.ReadWrite |
+| [group](../resources/group.md) | Group.ReadWrite.All | Not supported | Group.ReadWrite.All |
+| [group event](../resources/event.md) | Group.ReadWrite.All | Not supported | Not supported |
+| [group post](../resources/post.md) | Group.ReadWrite.All | Not supported | Group.ReadWrite.All |
+| [message](../resources/message.md) | Mail.ReadWrite | Mail.ReadWrite | Mail.ReadWrite | 
+| [organization](../resources/organization.md) | Directory.AccessAsUser.All | Not supported | Not supported |
+| [personal contact](../resources/contact.md) | Contacts.ReadWrite | Contacts.ReadWrite | Contacts.ReadWrite |
+| [user](../resources/user.md) | Directory.AccessAsUser.All | User.ReadWrite | User.ReadWrite.All |
```
Comment (dkershaw10) [inline api-reference/v1.0/api/opentypeextension_delete.md]: Is this last one true?  I find that hard to believe.  I recollect something about User.ReadWrite not allowing extension updates, but User.ReadWrite.All should.  I would change Directory.AccessAsUser.All -> User.ReadWrite.All.

File: api-reference/v1.0/api/opentypeextension_get.md [hunk 3]
```diff
@@ -12,15 +12,20 @@ The following table lists the three scenarios where you can get an open extensio
...
-| [Group post](../resources/post.md) | Group.Read.All | [Message](../resources/message.md) | Mail.Read |
-| [Organization](../resources/organization.md) | Directory.Read.All | [Personal contact](../resources/contact.md) | Contacts.Read |
-| [User](../resources/user.md) | User.ReadBasic.All | | |
+| [device](../resources/device.md) | Directory.Read.All | Not supported | Device.ReadWrite.All |
+| [event](../resources/event.md) | Calendars.Read | Calendars.Read | Calendars.Read |
+| [group](../resources/group.md) | Group.Read.All | Not supported | Group.Read.All |
+| [group event](../resources/event.md) | Group.Read.All | Not supported | Not supported |
+| [group post](../resources/post.md) | Group.Read.All | Not supported | Group.Read.All |
+| [message](../resources/message.md) | Mail.Read | Mail.Read | Mail.Read | 
+| [organization](../resources/organization.md) | Directory.AccessAsUser.All | Not supported | Not supported |
```
Comment (dkershaw10) [inline api-reference/v1.0/api/opentypeextension_get.md]: I believe that User.Read allows reading of basic Organization information.  That might also allow reading extension data. @pavankompelli Do you know if User.Read is sufficient here?

File: api-reference/v1.0/api/opentypeextension_get.md [hunk 4]
```diff
@@ -12,15 +12,20 @@ The following table lists the three scenarios where you can get an open extensio
...
+| [personal contact](../resources/contact.md) | Contacts.Read | Contacts.Read | Contacts.Read |
+| [user](../resources/user.md) | User.ReadBasic.All | User.ReadBasic.All | User.Read.All |
```
Comment (dkershaw10) [inline api-reference/v1.0/api/opentypeextension_get.md]: User.ReadBasic.All -> User.Read for both.
Definitely true for Personal Accounts, as you can only read your own profile and no one else's.  User.ReadBasic.All is not valid in consumer scenarios.

File: api-reference/v1.0/api/opentypeextension_post_opentypeextension.md [hunk 5]
```diff
@@ -6,15 +6,19 @@ Create an open extension ([openTypeExtension](../resources/openTypeExtension.md)
 
 ## Permissions
 
-One of the following permissions is required to call this API, depending on the resource you're creating the extension in. To learn more, including how to choose permissions, see [Permissions](../../../concepts/permissions_reference.md).
+Depending on the resource you're creating the extension in and the permission type (delegated or application) requested, the specified permission in the following table is the mimimum required to call this API. To learn more, including how to choose permissions, see [Permissions](../../../concepts/permissions_reference.md).
 
-|**Supported resource**|**Permission**|**Supported resource**|**Permission** |
+| Supported resource | Delegated (work or school account) | Delegated (personal Microsoft account) | Application |
```
Comment (dkershaw10) [inline api-reference/v1.0/api/opentypeextension_post_opentypeextension.md]: The permissions should be the same as for the delete topic

File: api-reference/v1.0/api/opentypeextension_update.md [hunk 6]
```diff
@@ -10,15 +10,19 @@ The data in an extension can be primitive types, or arrays of primitive types.
 
 ## Permissions
 
-One of the following permissions is required to call this API, depending on the resource that the extension was created in. To learn more, including how to choose permissions, see [Permissions](../../../concepts/permissions_reference.md).
+Depending on the resource that the extension was created in and the permission type (delegated or application) requested, the specified permission in the following table is the mimimum required to call this API. To learn more, including how to choose permissions, see [Permissions](../../../concepts/permissions_reference.md).
 
-|**Supported resource**|**Permission**|**Supported resource**|**Permission** |
+| Supported resource | Delegated (work or school account) | Delegated (personal Microsoft account) | Application |
```
Comment (dkershaw10) [inline api-reference/v1.0/api/opentypeextension_update.md]: The permissions should be the same as for the delete (and post) topics

File: api-reference/v1.0/api/opentypeextension_delete.md (hunk 1 above)
Comment (angelgolfer-ms) [reply api-reference/v1.0/api/opentypeextension_delete.md]: I'll change that.

File: api-reference/v1.0/api/opentypeextension_get.md (hunk 3 above)
Comment (pavankompelli) [reply api-reference/v1.0/api/opentypeextension_get.md]: If the actual object is returned from AAD, extensions will be returned as well (no further checks on read operation). So you are right.

Comment (angelgolfer-ms) [reply api-reference/v1.0/api/opentypeextension_get.md]: Ok, I will change the organization row to "User.Read".

File: api-reference/v1.0/api/opentypeextension_delete.md (hunk 2 above)
Comment (pavankompelli) [reply api-reference/v1.0/api/opentypeextension_delete.md]: Right, I think we debated whether extensions should be modeled after mobilenumber or displayName and finally went with displayName. So you are right.

Comment (angelgolfer-ms) [reply api-reference/v1.0/api/opentypeextension_delete.md]: I'll change the user row delegated (work/school) value to User.ReadWrite.All then.

**Comment by jthake-msft** [comment]
### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[api-reference/beta/api/opentypeextension_delete.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/ach-mg-perms/api-reference/beta/api/opentypeextension_delete.md) | :white_check_mark:Succeeded | |
[api-reference/beta/api/opentypeextension_get.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/ach-mg-perms/api-reference/beta/api/opentypeextension_get.md) | :white_check_mark:Succeeded | |
[api-reference/beta/api/opentypeextension_post_opentypeextension.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/ach-mg-perms/api-reference/beta/api/opentypeextension_post_opentypeextension.md) | :white_check_mark:Succeeded | |
[api-reference/beta/api/opentypeextension_update.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/ach-mg-perms/api-reference/beta/api/opentypeextension_update.md) | :white_check_mark:Succeeded | |
[api-reference/v1.0/api/opentypeextension_delete.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/ach-mg-perms/api-reference/v1.0/api/opentypeextension_delete.md) | :white_check_mark:Succeeded | |
[api-reference/v1.0/api/opentypeextension_get.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/ach-mg-perms/api-reference/v1.0/api/opentypeextension_get.md) | :white_check_mark:Succeeded | |
[api-reference/v1.0/api/opentypeextension_post_opentypeextension.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/ach-mg-perms/api-reference/v1.0/api/opentypeextension_post_opentypeextension.md) | :white_check_mark:Succeeded | |
[api-reference/v1.0/api/opentypeextension_update.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/ach-mg-perms/api-reference/v1.0/api/opentypeextension_update.md) | :white_check_mark:Succeeded | |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2018%5C11%5C7%5C77e68b91-cf14-3558-43ca-0d6d735c492d%5CPullRequest%5C201811070056331979-3272%5Cworkflow_report.html?sv=2015-02-21&sr=b&sig=ZIqoIBpfJq08kem8BqORMxZUmskIzzx5oDKhwUZ4N6g%3D&st=2018-11-07T00%3A54%3A20Z&se=2018-12-08T00%3A59%3A20Z&sp=r).

**Note:** If you changed an existing file name or deleted a file, broken links in other files to the deleted or renamed file are listed only in the full build report.

**Pull request merged**